modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeReplacementCheck.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.portal.json.JSONArrayImpl;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.io.InputStream;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Nícolas Moura
+ */
+public class UpgradeReplacementCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		String newContent = content;
+
+		JSONArray jsonArray = _getReplacementPatternsJSONArray(
+			"replacement-patterns.json");
+
+		for (int i = 0; i < jsonArray.length(); i++) {
+			JSONObject jsonObject = jsonArray.getJSONObject(i);
+
+			if (!_isValidExtension(fileName, jsonObject)) {
+				continue;
+			}
+
+			boolean replaced = false;
+
+			Pattern pattern = Pattern.compile(jsonObject.getString("regEx"));
+
+			Matcher matcher = pattern.matcher(content);
+
+			while (matcher.find()) {
+				String methodCall = matcher.group();
+
+				if (fileName.endsWith(".java")) {
+					String className = jsonObject.getString("className");
+
+					if (!className.equals("") && !className.endsWith("Util") &&
+						!hasClassOrVariableName(
+							className, content, content, methodCall)) {
+
+						continue;
+					}
+				}
+
+				newContent = StringUtil.replace(
+					newContent, methodCall,
+					StringUtil.replace(
+						methodCall, jsonObject.getString("oldSub"),
```

Comment:
@nicolas-sss, what `oldSub` means? Can we have a more meaningful name?

Reply:
`oldSub` is the old string is the exact name of the `StringUtil.replace `parameter

Reply:
I see, @nicolas-sss. How about we rename it? What do you think about the following?
```
{
	"className": "",
	"newImports": [
	],
	"newReference": "",
	"pattern": "getCode",
	"patternReplacement": "getRegionCode",
	"regex": "\\w+\\.\\w*\\(?\\s*\\)?\\s*\\.?getCode\\(",
	"validExtensions": [
		"java"
	]
}
```

Reply:
pattern has occurrences in the code already, it can get confusing. What do you think of `original` and `originalReplacement`